### ISSUE TITLE
Remove returned pointers to primitive types from the generated C#

### ISSUE THREAD

**Issue by ddobrev** [mono/CppSharp#853]
This can be achieved using the new ref returns from C# 7. It's easy to generate the code but what's more problematic is the automatic compilation of the generated code. There are 3 options there:

1. Force users to upgrade to VS 2017 (requires https://github.com/mono/CppSharp/issues/785 to be fixed first);
2. Have users install just https://www.nuget.org/packages/Microsoft.Net.Compilers/ replacing their compilers;
3. Ship the entire compiler along with C++#.

All of these have serious flaws so a decision must be taken in advance.

**Comment by tritao** [comment]
Good idea Dimitar, as an alternative to the options you provided we could also keep the old code path and use it if the user is targeting < VS 2017.
